internal/proposals/migration-hooks/migration-hooks-proposal.md

```diff
@@ -0,0 +1,233 @@
+
+// Define type for jobs
+type job interface {
+	execute() error
+	when() (component, targetVersion, executionTime, installationType)
+}
+
+func initializeMaps() error {
+	for _, job := range jobs {
+		// TODO: Add job to corresponding map
+	}
+}
+
+// Register job
+func register(j job) {
+	jobs = append(jobs, job)
+}
+
+// Gets called in deletion.go and deployment.go when `NewDeletion`/`NewDeployment` is being called
+func SetKymaVersion(version targetVersion) {
+	kymaVersion = version
+	if err := initializeMaps(); err != nil {
+		// TODO: handle error
+	}
+}
+
+// Function should be called before component is being deployed/upgraded
+func ExecutePre(component string, it installationType) {
+	start := time.Now()
+	// TODO: Executes the registered functions for given component; using maps
+	//       If map for given key(aka component) is empty, nothing will be done
+	//       Check installationType, to know which map should be used
+	t := time.Now()
+	duration += t.Sub(start)
+}
+
+// Function should be called after compoent is being deployed/upgraded
+func ExecutePost(component string, it installationType) {
+	start := time.Now()
+	// TODO: Executes the registered functions for given component; using maps
+	//       If map for given key(aka component) is empty, nothing will be done
+	//       Check installationType, to know which map should be used
+	t := time.Now()
+	duration += t.Sub(start)
+}
+
+// Returns duration of all jobs for benchmarking
+func GetDuration() time.Duration {
+	return duration
+}
+
+```
+
+### component1.go - Exmaple component file draft
+
+```go
+package jobManager
+
+// Register job using implemented Interface
+register(job1)
+type job1 struct{}
+
+func (j job1) execute() {
+	// Do something
+  ...
+  return nil
+}
+
+func (j job1) when() {
+	return ("kiali", "1.22", Pre, Deploy)
+}
+
+```
+
+### Hook for global jobs in `hydroform/parallel-install/deployment.go`; Pre/Post global Jobs - Deploy
+
+Pre- and post-jobs will be executed before and after Kyma deploy.
+
+```go
+import "hydroform/parallel-install/jobs"
+func (i *Deployment) deployComponents(ctx context.Context, cancelFunc context.CancelFunc, phase InstallationPhase, eng *engine.Engine, cancelTimeout time.Duration, quitTimeout time.Duration) error {
+  ...
+  deploymentJobs.ExecutePre("global", jobManager.Deploy)
+  statusChan, err := eng.Deploy(ctx)
+  ...
+  // for-Loop for component install
+  ...
+  deploymentJobs.ExecutePost("global", jobManager.Deploy) 
+}
+```
+
+### Hook for component jobs in `hydroform/parallel-install/engine.go`; Pre/Post Component Jobs - Deploy and Deletion 
+
+Pre- and post-jobs will be executed before and after each Kyma component. In this way only the componentns which will be installed/upgraded will be considered by the JobManager &#8594; Other will just not be executed. Furthermore, since the pre-requisites of components are deployed using the `worker` function of an `engine` as well, the JobManager handles them automatically.
```

Comment:
```suggestion
Pre- and post-jobs will be executed before and after each Kyma component. In this way, only the components that will be installed or upgraded are considered by the JobManager &#8594; others are not executed. Furthermore, since the prerequisites of components are deployed using the `worker` function of an `engine` as well, the JobManager handles them automatically.
```